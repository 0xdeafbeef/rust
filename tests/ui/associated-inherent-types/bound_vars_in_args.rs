#![feature(non_lifetime_binders, inherent_associated_types)]
#![expect(incomplete_features)]

// Test that we can resolve to the right IAT when the self type
// contains a bound type.

struct Foo<T: ?Sized>(T);

impl Foo<[u8]> {
    type IAT = u8;
}

impl<T: Sized> Foo<T> {
    type IAT = u8;
}

struct Bar
//~^ ERROR: the size for values of type `T` cannot be known at compilation time
//~| ERROR: the size for values of type `T` cannot be known at compilation time
where
    for<T> Foo<T>::IAT: Sized;

fn main() {}
