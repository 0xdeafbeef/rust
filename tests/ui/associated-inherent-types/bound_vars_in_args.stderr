error[E0277]: the size for values of type `T` cannot be known at compilation time
  --> $DIR/bound_vars_in_args.rs:17:1
   |
LL | struct Bar
   | ^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `T`
note: required by a bound in `Foo<T>::IAT`
  --> $DIR/bound_vars_in_args.rs:13:9
   |
LL | impl<T: Sized> Foo<T> {
   |         ^^^^^ required by this bound in `Foo<T>::IAT`
LL |     type IAT = u8;
   |          --- required by a bound in this associated type

error[E0277]: the size for values of type `T` cannot be known at compilation time
  --> $DIR/bound_vars_in_args.rs:17:1
   |
LL | / struct Bar
LL | |
LL | |
LL | | where
LL | |     for<T> Foo<T>::IAT: Sized;
   | |______________________________^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `T`
note: required by a bound in `Foo<T>::IAT`
  --> $DIR/bound_vars_in_args.rs:13:9
   |
LL | impl<T: Sized> Foo<T> {
   |         ^^^^^ required by this bound in `Foo<T>::IAT`
LL |     type IAT = u8;
   |          --- required by a bound in this associated type

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
